spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java

```diff
@@ -263,20 +264,40 @@ private String determinePlatformName(final String platformName) {
 	}
 
 	private void validateAllAppsRegistered(StreamDeploymentRequest streamDeploymentRequest) {
+		StreamDefinition streamDefinition = this.streamDefinitionRepository.findOne(streamDeploymentRequest.getStreamName());
 		for (AppDeploymentRequest adr : streamDeploymentRequest.getAppDeploymentRequests()) {
 			String version = ResourceUtils.getResourceVersion(adr.getResource());
-			validateAppVersionIsRegistered(adr, version);
+			validateAppVersionIsRegistered(getRegisteredName(streamDefinition, adr.getDefinition().getName()), adr, version);
 		}
 	}
 
-	public void validateAppVersionIsRegistered(AppDeploymentRequest appDeploymentRequest, String appVersion) {
-		String name = appDeploymentRequest.getDefinition().getName();
+	private String getRegisteredName(StreamDefinition streamDefinition, String adrAppName) {
+		for (StreamAppDefinition appDefinition: streamDefinition.getAppDefinitions()) {
+			if (appDefinition.getName().equals(adrAppName)) {
+				return appDefinition.getRegisteredAppName();
+			}
+		}
+		return adrAppName;
+	}
+
+	public void validateAppVersionIsRegistered(StreamDefinition streamDefinition, AppDeploymentRequest appDeploymentRequest, String appVersion) {
+		String registeredAppName = getRegisteredName(streamDefinition, appDeploymentRequest.getDefinition().getName());
+		String appTypeString = appDeploymentRequest.getDefinition().getProperties()
+				.get(DataFlowPropertyKeys.STREAM_APP_TYPE);
+		ApplicationType applicationType = ApplicationType.valueOf(appTypeString);
+		if (!this.appRegistryService.appExist(registeredAppName, applicationType, appVersion)) {
+			throw new IllegalStateException(String.format("The %s:%s:%s app is not registered!",
+					registeredAppName, appTypeString, appVersion));
+		}
+	}
+
+	public void validateAppVersionIsRegistered(String registeredAppName, AppDeploymentRequest appDeploymentRequest, String appVersion) {
```

Comment:
This method is only used by the private `validateAllAppsRegistered` and should be private as well. Furthermore it can be reused by the `public void validateAppVersionIsRegistered(StreamDefinition streamDefinition, AppDeploymentRequest appDeploymentRequest, String appVersion)`